packages/zendesk-adapter/src/filters/guide_theme.ts

```diff
@@ -0,0 +1,139 @@
+import { logger } from '@salto-io/logging'
+import { collections } from '@salto-io/lowerdash'
+import JSZip from 'jszip'
+import _, { remove } from 'lodash'
+import { FETCH_CONFIG, isGuideEnabled, isGuideThemeEnabled } from '../config'
+import {
+  BRAND_TYPE_NAME,
+  GUIDE_THEME_TYPE_NAME,
+  ZENDESK,
+} from '../constants'
+import { FilterCreator } from '../filter'
+import { download } from './guide_themes/download'
+
+const log = logger(module)
+const { awu } = collections.asynciterable
+
+type ThemeFile = { filename: string; content: StaticFile }
+type ThemeDirectory = { [key: string]: ThemeFile | ThemeDirectory }
+
+const addFileToDirectory = (root: ThemeDirectory, relativeFilename: string, file: ThemeFile): void => {
+  const pathSegments = relativeFilename.split('/')
+  const fileSegment = pathSegments.pop() as string // Remove and store the file segment, the array is never empty
+
+  // Use reduce to traverse and/or build the directory structure
+  const targetDirectory = pathSegments.reduce((currentDirectory, segment) => {
+    if (!currentDirectory[segment] || typeof currentDirectory[segment] !== 'object') {
+      currentDirectory[segment] = {}
+    }
+    return currentDirectory[segment] as ThemeDirectory
+  }, root)
+
+  // Add the file to the target directory
+  targetDirectory[fileSegment] = file
+}
+
+const unzipFolderToElements = async (buffer: Buffer, brandName: string, name: string): Promise<ThemeDirectory> => {
+  const zip = new JSZip()
+  const unzippedContents = await zip.loadAsync(buffer)
+
+  const elements: ThemeDirectory = {}
+  await Promise.all(Object.entries(unzippedContents.files).map(async ([relativePath, file]) => {
+    if (!file.dir) {
+      const content = await file.async('nodebuffer')
+      addFileToDirectory(elements, relativePath, {
+        filename: relativePath,
+        content: new StaticFile({ filepath: `${ZENDESK}/themes/brands/${brandName}/${name}/${relativePath}`, content }),
+      })
+    }
+  }))
+  return elements
+}
+
+const getFullName = (instance: InstanceElement): string => instance.elemID.getFullName()
+
+const addDownloadErrors = (
+  theme: InstanceElement,
+  downloadErrors: string[]
+): SaltoError[] => ((downloadErrors.length > 0)
+  ? downloadErrors.map(e => ({
+    message: `Error fetching theme id ${theme.value.id}, ${e}`,
+    severity: 'Warning',
+  }))
+  : [{
+    message: `Error fetching theme id ${theme.value.id}, no content returned from Zendesk API`,
+    severity: 'Warning',
+  }])
+
+/**
+ * Fetches guide theme content
+ */
+const filterCreator: FilterCreator = ({ config, client }) => ({
+  name: 'guideThemesFilter',
+  onFetch: async elements => {
+    if (!isGuideEnabled(config[FETCH_CONFIG]) || !isGuideThemeEnabled(config[FETCH_CONFIG])) {
+      return undefined
+    }
+
+    const instances = elements.filter(isInstanceElement)
+    const guideThemes = instances.filter(i => i.elemID.typeName === GUIDE_THEME_TYPE_NAME)
```

Comment:
nit: 
```suggestion
    const guideThemes = instances.filter(inst => inst.elemID.typeName === GUIDE_THEME_TYPE_NAME)
```
(we try to avoid single-letter / unclear acronyms)